File: avalon-modules/avalon-authorization/src/main/java/io/github/lishangbu/avalon/authorization/service/OauthRegisteredClientService.java
package io.github.lishangbu.avalon.authorization.service;

import io.github.lishangbu.avalon.authorization.entity.OauthRegisteredClient;
import java.util.List;
import java.util.Optional;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/// OAuth2 注册客户端服务
///
/// 提供 OAuth2 注册客户端的增删改查能力
///
/// @author lishangbu
/// @since 2026/3/19
public interface OauthRegisteredClientService {

    /// 根据条件分页查询注册客户端。
    Page<OauthRegisteredClient> getPageByCondition(
            OauthRegisteredClient registeredClient, Pageable pageable);

    /// 根据条件查询注册客户端列表。
    List<OauthRegisteredClient> listByCondition(OauthRegisteredClient registeredClient);

    /// 根据 ID 查询注册客户端。
    Optional<OauthRegisteredClient> getById(String id);

    /// 新增注册客户端。
    OauthRegisteredClient save(OauthRegisteredClient registeredClient);

    /// 更新注册客户端。
    OauthRegisteredClient update(OauthRegisteredClient registeredClient);

    /// 根据 ID 删除注册客户端。
    void removeById(String id);
}

File: avalon-modules/avalon-authorization/src/main/kotlin/io/github/lishangbu/avalon/authorization/service/OauthRegisteredClientService.kt
package io.github.lishangbu.avalon.authorization.service

import io.github.lishangbu.avalon.authorization.entity.dto.OauthRegisteredClientSpecification
import io.github.lishangbu.avalon.authorization.entity.dto.OauthRegisteredClientView
import io.github.lishangbu.avalon.authorization.entity.dto.SaveOauthRegisteredClientInput
import io.github.lishangbu.avalon.authorization.entity.dto.UpdateOauthRegisteredClientInput
import org.babyfish.jimmer.Page
import org.springframework.data.domain.Pageable

/**
 * OAuth2 注册客户端服务
 *
 * 定义 OAuth2 注册客户端的管理能力
 *
 * @author lishangbu
 * @since 2026/3/19
 */
interface OauthRegisteredClientService {
    /** 按条件分页查询 OAuth2 注册客户端 */
    fun getPageByCondition(
        specification: OauthRegisteredClientSpecification,
        pageable: Pageable,
    ): Page<OauthRegisteredClientView>

    /** 按条件查询 OAuth2 注册客户端列表 */
    fun listByCondition(specification: OauthRegisteredClientSpecification): List<OauthRegisteredClientView>

    /** 按 ID 查询 OAuth2 注册客户端 */
    fun getById(id: String): OauthRegisteredClientView?

    /** 保存 OAuth2 注册客户端 */
    fun save(command: SaveOauthRegisteredClientInput): OauthRegisteredClientView

    /** 更新 OAuth2 注册客户端 */
    fun update(command: UpdateOauthRegisteredClientInput): OauthRegisteredClientView

    /** 按 ID 删除 OAuth2 注册客户端 */
    fun removeById(id: String)
}
